### DISCUSSION TITLE
Relaxation time when using bantime.increment

### DISCUSSION THREAD

**Discussion by madhatter0** [fail2ban/fail2ban#3863] [Q&A] [answered]
I'm using incremental banning, with an initial period of 10m, and most of the defaults.  The config is this:

```
[smtp-auth]
enabled  = true
filter   = smtp-auth
action   = iptables-multiport[name=smtp, port="smtp,submission", protocol=tcp]
logpath  = /var/log/mail.log
maxretry = 3
bantime  = 600
bantime.increment = true
#bantime.factor = 1
bantime.maxtime = 24h
```

 So if an IP gets a ban, waits 15 minutes, then tries again, (s)he gets detected, and I see something like this in the logs:
```

2024-10-18 09:24:31,611 fail2ban.observer       [1893454]: INFO    [smtp-auth] IP 192.0.2.4 is bad: 1 # last 2024-10-18 09:1
3:33 - incr 10m to 20m
2024-10-18 09:24:31,612 fail2ban.observer       [1893454]: NOTICE  [smtp-auth] Increase Ban 192.0.2.4 (2 # 20m -> 2024-10-18 09:44:31)
```

This IP is fairly annoying, and continues to rack up bans, until (s)he hits the (configured) maximum of one day:

```
2024-10-21 10:13:10,270 fail2ban.observer       [1897473]: INFO    [smtp-auth] IP 192.0.2.4 is bad: 8 # last 2024-10-20 12:50:04 - incr 10m to 1d
2024-10-21 10:13:10,271 fail2ban.observer       [1897473]: NOTICE  [smtp-auth] Increase Ban 192.0.2.4 (9 # 1d -> 2024-10-22 10:13:09)
```

After an incremented ban elapses, there is some period of time when a further log detection is treated as reoffending, and an incremental ban is triggered.  But presumably if a long enough period elapses, a further log detection would be treated as a fresh offence, and the ban would start again at 10m?

Is that presumption correct, and if so, what's the period?  Is it the same as the length of ban most recently served?  Or something else?  Or am I completely wrong, and a once-bad IP is bad forever?

**Comment by sebres** [comment, marked as answer]
> Is that presumption correct, and if so, what's the period?

The known as bad IPs are stored in fail2ban database, which will be periodically purged and the algorithm is described in https://github.com/fail2ban/fail2ban/issues/2428#issuecomment-494865081

For your example IP, banned 9th time for 1 day (and default `dbpurgeage`, also 1 day), the IP will remain for next 3 days after it'd become banned last time. If during his period it'd be banned again (gets its bantime `2d`), it'd be purged not earlier than in 6 days.
You can also increase `dbpurgeage` (in `fail2ban.local`) to avoid too early purge for all IPs (e. g. `10d` will hold all IPs at least 10 days)... But I would not set it too large, just to avoid retaining a lot of dynamic IPs for a long time.
